### PR TITLE
Fix #448: Add params for redis multi-AZ support

### LINKED ISSUE #448: Redis: Multi-AZ support
It would be nice if `convox services create redis` supported [multiple AZ replication groups](https://aws.amazon.com/blogs/aws/elasticache-redis-multi-az/).

The [existing template](https://github.com/convox/rack/blob/master/api/models/templates/service/redis.tmpl) is already using the `ReplicationGroup` resource, so I think adding the following parameters would enable support (and keep existing functionality as-is):

* [`AutomaticFailoverEnabled`](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-replicationgroup.html#cfn-elasticache-replicationgroup-automaticfailoverenabled): boolean, default to `False`
* [`NumCacheClusters `](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-elasticache-replicationgroup.html#cfn-elasticache-replicationgroup-numcacheclusters) integer, default to `1`

Validation of parameter combinations should rely on CloudFormation.

### PR DESCRIPTION
**Pull request by mwarkentin**
Fixes #448 

Add params:

* `AutomaticFailoverEnabled` - default: No
* `NumCacheClusters` - default: 1

## Release Playbook
- [ ] Rebase against master
- [ ] Pass checks
- [ ] Release branch ()
- [ ] Pass CI ()
- [ ] Code review
- [ ] Merge into master
- [ ] Release master ()
- [ ] Pass CI ()
- [ ] Update dev and testing racks
- [ ] Publish release
- [ ] Release CLI

### CODE REVIEW

File: api/models/templates/service/redis.tmpl [hunk 1]
```diff
@@ -2,6 +2,11 @@
   {
     "AWSTemplateFormatVersion" : "2010-09-09",
     "Parameters": {
+      "AutomaticFailoverEnabled": {
+        "Type": "String",
```
Comment (ddollar) [inline api/models/templates/service/redis.tmpl]: Can we make this parameter be Yes/No like other booleans?

https://github.com/convox/rack/blob/master/api/models/templates/app.tmpl#L37-L38

Comment (mwarkentin) [inline api/models/templates/service/redis.tmpl]: Sure thing - first example of a bool I came across was https://github.com/convox/rack/blob/master/api/models/templates/service/mysql.tmpl#L20-L23, so I just followed the same pattern.. will update.

Comment (mwarkentin) [inline api/models/templates/service/redis.tmpl]: @ddollar Done.

**Comment by nzoschke** [comment]
This is great, and fits in with the work here: https://github.com/convox/rack/pull/428

Good catch on the mysql bool. Yes/No is the better convention, and we can fix that up on mysql too.

**Comment by nzoschke** [comment]
Actually both postgres and mysql both use "true" and "false" and pass it straight through as `"MultiAZ": { "Ref": "MultiAZ" },`

It's not clear if "Yes" and "No" would have the same effect.

**Comment by nzoschke** [comment]
"Yes" and "No" do not work:

```
19:13:25 UTC-0700	CREATE_FAILED	AWS::RDS::DBInstance	Instance	Encountered unsupported boolean value Yes
```

**Comment by nzoschke** [comment]
Also caught this error:

>>> Automatic failover requires 2 or more cache clusters. Create more cache clusters or disable automatic failover.

**Comment by nzoschke** [comment]
And this error:

> Automatic failover is not supported for T1 and T2 cache node types.

**Comment by mwarkentin** [comment]
@nzoschke Do you prefer I revert or do a `reset --hard` to switch back to `true`/`false`?

Also, I discussed w/ @ddollar, and he said to rely on the CFN errors (which I believe is what you're posting) - are you just documenting your testing right now or are these comments things which need to get fixed?

Thanks for looking at this!

**Comment by nzoschke** [comment]
Just documenting my observations right now. I am planning on cherry-picking just 7b4e829ac76b6516db7857648304b6dc29f2fe9b onto https://github.com/convox/rack/pull/428/files and working there.

**Comment by mwarkentin** [comment]
Perfect - let me know when you've done that and I'll close this.

**Comment by mwarkentin** [comment]
Closing in favour of #428.

**Pull request closed without merging**